agent/comms.go

```diff
@@ -27,8 +28,12 @@ func (a *orbAgent) connect(ctx context.Context, config config.MQTTConfig) (mqtt.
 	})
 	opts.SetConnectionLostHandler(func(client mqtt.Client, err error) {
 		a.logger.Error("connection to mqtt lost", zap.Error(err))
+		// If it is a bug on the mqttclient, stop the agent
+		if strings.Contains(err.Error(), "BUG") {
+			a.Stop(ctx)
+		}
 		a.logger.Info("reconnecting....")
-		client.Connect()
+		a.requestReconnection(ctx, a.client, config)
```

Comment:
`a.Stop(ctx)` stops all configured backends, don't we need to start the agent again in this scenario to setup backend to the initial state?

Reply:
the a.Stop is a termination command, I can leave it explicitly with a return command there!